projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -57,15 +57,46 @@
 int32_t i2c_init(struct i2c_desc **desc,
 		 const struct i2c_init_param *param)
 {
-	if (desc) {
-		// Unused variable - fix compiler warning
-	}
+	i2c_desc *dev;
+	int32_t ret;
 
-	if (param->type) {
-		// Unused variable - fix compiler warning
-	}
+	dev = calloc(1, sizeof *dev);
+	if(!dev)
+		return FAILURE;
+
+	dev->type = param->type;
+	dev->id = XPAR_AXI_IIC_MAIN_DEVICE_ID;
+	dev->max_speed_hz = param->max_speed_hz;
+	dev->slave_address = param->slave_address;
+
+	dev->config = XIic_LookupConfig(dev->id);
+	if (dev->config == NULL)
+		goto error;
+
+	ret = XIic_CfgInitialize(&dev->instance, dev->config,
+				 dev->config->BaseAddress);
+	if(ret != XST_SUCCESS)
+		goto error;
+
+	Xil_ExceptionRegisterHandler(XIL_EXCEPTION_ID_IRQ_INT,
+				     XIic_InterruptHandler, &dev->instance);
+
```

Comment:
We have to make sure that this is the right approach (e.g.,  shouldn't we call Xil_ExceptionInit() in advance and Xil_ExceptionEnable() after? ).

Reply:
This document on page 70 notes that for ARM Cortex-A53, Cortex-R5, and Cortex-A9 the exception handlers are being initialized statically and this function is only used for backwards compatibility. Should we add it to have this feature as well?

https://github.com/Xilinx/embeddedsw/blob/master/lib/bsp/standalone/doc/standalone.pdf

Reply:
I will add Xil_ExceptionInit() just to eliminate unnecessary risks. I would add a function call and a return, but the application has no constrains on speed that will be affected by this.